src/spec.c

```diff
@@ -366,6 +366,13 @@ size_t IndexSpec_TotalMemUsage(IndexSpec *sp, size_t doctable_tm_size, size_t ta
   return res;
 }
 
+
+void IndexSpec_SetIndexErrorMessage(IndexSpec *sp, const char *error, RedisModuleString *key) {
+  RedisModule_Assert(sp);
+  RedisModule_Assert(error);
+  IndexError_AddError(&sp->stats.indexError, error, key);
```

Comment:
Are these assertions temporary? We will get SEGV if `sp` is not set.

Reply:
if sp is not set, we really messed up